lib/ldp.js

```diff
@@ -422,9 +422,21 @@ class LDP {
     }
   }
 
+  // delete resource with acl and meta links
   async deleteResource (path) {
+    const linkPath = this.resourceMapper._removeDollarExtension(path)
     try {
-      return await withLock(path, { mustExist: false }, () => promisify(fs.unlink)(path))
+      // first delete file, then links with write permission only (atomic delete)
+      await withLock(path, { mustExist: false }, () => promisify(fs.unlink)(path))
```

Comment:
If this is a folder, it seems that the ACL/meta deletion would fail?
I would expect this statement to come last?

Reply:
This is never a folder. 

`async delete (url)` is calling : 
- `await deleteContainer (directory)`
- or` await deleteResource (path)`

The PR changes only  `async deleteResource (path)` and `path` is always a `file`

Reply:
OK, shall we change the name of this method then to be `deleteDocument` so that this is clear?

Reply:
May be a good idea `document` can be understood as `resource and links`.